### ISSUE TITLE
Restructure repository branches

### ISSUE THREAD

**Issue by mmarchini** [nodejs/node-v8#170]
- [ ] Create `main` branch in this repository, set it as default branch
- [ ] Keep `canary` branch as we have today
  - [ ] Alternatively have a `canary/` folder which is a copy of `nodejs/node` with patches applied
  - [ ] Alternatively 2 keep the V8 canary as a `v8-canary` branch on `nodejs/node` and have it as a submodule here
- [ ] Ensure Jenkins fetches the `canary` branch for relevant jobs
- [ ] Use GitHub Actions to update the `canary` branch
- [ ] Drop `canary-base` branch on `nodejs/node` and use patch files on this repository instead

**Comment by mmarchini** [comment]
We can start by creating the `main` branch with an empty README, and then slowly we can work on the other changes there (via PRs) and then once we're confident this new system is working and is better than the current state we can switch it.

**Comment by targos** [comment]
> Drop `canary-base` branch on `nodejs/node` and use patch files on this repository instead.

I'm not opposed to it, but having a branch to manage the canary patches helps me a lot because I can easily use `git cherry-pick` and `git rebase`. We need to make sure patch files don't slow down the regular processes of
- keeping gyp files up to date
- preparing the V8 update PRs in `nodejs/node`

**Comment by mmarchini** [comment]
Definitely. I'm proposing these changes because I do believe they will make the overall process _faster_ (or at least _simpler_), but we'll definitely need some experimentation in order to get it right.

**Comment by mmarchini** [comment]
Besides dropping the `canary-base` branch, what are your thoughts on the other changes?

**Comment by targos** [comment]
Other changes SGTM
